src/main/java/seedu/address/model/task/UniqueTodoList.java

```diff
@@ -0,0 +1,148 @@
+        FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent task as the given argument.
+     */
+    public boolean contains(Task toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameTask);
+    }
+
+    /**
+     * Adds a task to the list.
+     * The task must not already exist in the list.
+     */
+    public void add(Task toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateTaskException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Replaces the task {@code target} in the list with {@code editedTask}.
+     * {@code target} must exist in the list.
+     * The task's name, type, and date of {@code editedTask} must not be the same as another
+     * existing task in the list.
+     */
+    public void setTask(Task target, Task editedTask) {
+        requireAllNonNull(target, editedTask);
+
+        int index = internalList.indexOf(target);
+        if (index == -1) {
+            throw new TaskNotFoundException();
+        }
+
+        if (!target.isSameTask(editedTask) && contains(editedTask)) {
+            throw new DuplicateTaskException();
+        }
+
+        internalList.set(index, editedTask);
+    }
+
+    /**
+     * Removes the equivalent task from the list.
+     * The task must exist in the list.
+     */
+    public void remove(Task toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new TaskNotFoundException();
+        }
+    }
+
+    public void setTasks(UniqueTodoList replacement) {
+        requireNonNull(replacement);
+        internalList.setAll(replacement.internalList);
+    }
+
+    /**
+     * Replaces the contents of this list with {@code tasks}.
+     * {@code tasks} must not contain duplicate tasks.
+     */
+    public void setTasks(List<Task> tasks) {
+        requireAllNonNull(tasks);
+        if (!tasksAreUnique(tasks)) {
+            throw new DuplicateTaskException();
+        }
+
+        internalList.setAll(tasks);
+    }
+
+    /**
+     * Sort the list using the comparator.
+     *
+     * @param comparator comparator for sorting
+     */
+    public void sort(Comparator<? super Task> comparator) {
+        internalList.sort(comparator);
+    }
+
+    /**
+     * Returns the backing list as an unmodifiable {@code ObservableList}.
+     */
+    public ObservableList<Task> asUnmodifiableObservableList() {
+        return internalUnmodifiableList;
+    }
+
+    @Override
+    public Iterator<Task> iterator() {
+        return internalList.iterator();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+            || (other instanceof UniqueTodoList // instanceof handles nulls
+            && internalList.equals(((UniqueTodoList) other).internalList));
```

Comment:
Why does this work? Since the field internalList is private, how can u just compare using .internalList

Reply:
I think it should work since this is taken directly from our current UniqueModuleList class. In addition, field internalList is private means that it can only be accessed by the UniqueTodoList class and the code "((UniqueTodoList) other).internalList))" is inside the UniqueTodoList class. Hence, it is not a problem.